include/sacn/cpp/receiver.h

```diff
@@ -0,0 +1,392 @@
+/******************************************************************************
+ * Copyright 2020 ETC Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************
+ * This file is a part of sACN. For more information, go to:
+ * https://github.com/ETCLabs/sACN
+ *****************************************************************************/
+
+#ifndef SACN_CPP_RECEIVER_H_
+#define SACN_CPP_RECEIVER_H_
+
+/// @file sacn/cpp/receiver.h
+/// @brief C++ wrapper for the sACN Receiver API
+
+#include "sacn/receiver.h"
+#include "etcpal/cpp/inet.h"
+
+/// @defgroup sacn_receiver_cpp sACN Receiver API
+/// @ingroup sacn_cpp_api
+/// @brief A C++ wrapper for the sACN Receiver API
+
+namespace sacn
+{
+/// @ingroup sacn_receiver_cpp
+/// @brief An instance of sACN Receiver functionality.
+///
+/// CHRISTIAN TODO: FILL OUT THIS COMMENT MORE -- DO WE NEED A using_receiver.md???
+class Receiver
+{
+public:
+  /// A handle type used by the sACN library to identify receiver instances.
+  using Handle = sacn_receiver_t;
+  /// An invalid Handle value.
+  static constexpr Handle kInvalidHandle = SACN_RECEIVER_INVALID;
+
+  /// @ingroup sacn_receiver_cpp
+  /// @brief A base class for a class that receives notification callbacks from a sACN receiver.
+  class NotifyHandler
+  {
+  public:
+    virtual ~NotifyHandler() = default;
+
+    /// @brief Notify that one or more sources have been found.
+    /// @param handle Handle to the receiver instance for which sources were found.
+    /// @param found_sources Array of structs describing the source or sources that have been found with their current
```

Comment:
My thought was to avoid the unneeded copying, and it mirrors what we do in RDMnet.  In fact, a lot of these APIs mirror RDMnet... :)

Reply:
RDMnet sometimes takes the approach of letting the user pick one or the other. For example, the RptClientList class: https://github.com/ETCLabs/RDMnet/blob/develop/include/rdmnet/cpp/message_types/rpt_client.h#L87

Users can choose the raw array of C client entries, or choose to copy them into a vector for nicer syntax.

Reply:
Christian, I'm on the fence on this one.  I wanted to avoid copying, but it isn't as nice as a std::vector&. If you want to create a couple of wrapper classes for this, I'm not against it.  If it's ok with you two, I don't see a need for me to do anything more here.